src/lib/markets/graph.ts

```diff
@@ -983,3 +983,129 @@ export const getPriceAtVersion = async ({
 
   return res.marketVersionPrice?.price ?? 0n
 }
+
+export async function fetchTradeHistory({
+  graphClient,
+  address,
+  first,
+  offset,
+}: {
+  graphClient: GraphQLClient
+  address: Address
+  first: number
+  offset: number
+}) {
+  const tradeHistoryQuery = gql(`
+   query fetchTradeHistory($account: Bytes!, $first: Int!, $offset: Int!) {
+    accountPositionProcesseds(
+      where: { account: $account },
+      orderBy: toOracleVersion,
+      orderDirection: desc,
+      first: $first,
+      skip: $offset,
+    ) {
+      accumulationResult_collateralAmount, accumulationResult_keeper, accumulationResult_positionFee, priceImpactFee
+        accumulatedPnl, accumulatedFunding, accumulatedInterest, accumulatedMakerPositionFee, accumulatedValue
+        side, size, fromOracleVersion, toOracleVersion, toVersionPrice, toVersionValid, collateral, blockNumber,
+        market
+      update {
+        version, collateral, newMaker, newLong, newShort, valid, transactionHash, price, priceImpactFee,
+        localPositionId, globalPositionId, market, account, blockNumber, blockTimestamp, protect, interfaceFee, orderFee, side, delta
+      }
+    }
+   }
+  `)
+
+  const { accountPositionProcesseds } = await graphClient.request(tradeHistoryQuery, {
+    account: address,
+    first,
+    offset,
+  })
+
+  const entities = accountPositionProcesseds.reduce(
+    (acc, entity) => {
+      if (!acc[entity.market]) acc[entity.market] = []
+      acc[entity.market].push(entity)
+      return acc
+    },
+    {} as Record<string, typeof accountPositionProcesseds>,
```

Comment:
nit: Safer to make this a `Record<Address, ...>` and use `getAddress` on the `entity.market`